packages/web/src/hooks/useTrackCoverArt.ts

```diff
@@ -1,36 +1,108 @@
+import { useEffect } from 'react'
+
 import { imageBlank as imageEmpty } from '@audius/common/assets'
-import { useImageSize } from '@audius/common/hooks'
-import { SquareSizes, ID, CoverArtSizes } from '@audius/common/models'
-import { cacheTracksActions, cacheTracksSelectors } from '@audius/common/store'
+import { useImageSize2 } from '@audius/common/hooks'
+import { SquareSizes, ID } from '@audius/common/models'
+import { getDominantColorsByTrack } from '@audius/common/src/store/average-color/selectors'
+import { setDominantColors } from '@audius/common/src/store/average-color/slice'
+import { cacheTracksSelectors, CommonState } from '@audius/common/store'
 import { Maybe } from '@audius/common/utils'
 import { useDispatch } from 'react-redux'
 
+import { dominantColor } from 'utils/imageProcessingUtil'
 import { useSelector } from 'utils/reducer'
 
-const { fetchCoverArt } = cacheTracksActions
 const { getTrack } = cacheTracksSelectors
 
-export const useTrackCoverArt = (
-  trackId: number | null | string | undefined,
-  coverArtSizes: CoverArtSizes | null,
-  size: SquareSizes,
-  defaultImage: string = imageEmpty as string
-) => {
-  const dispatch = useDispatch()
-  return useImageSize({
-    dispatch,
-    id: trackId,
-    sizes: coverArtSizes,
-    size,
-    action: fetchCoverArt,
-    defaultImage
+export const useTrackCoverArt = ({
+  trackId,
+  size,
+  defaultImage
+}: {
+  trackId: Maybe<ID>
```

Comment:
I've been seeing this type helper more and more and I'm wondering why it's preferable to just making a field optional?

Reply:
idk... happy to update though haha